frontend/src/lib-components/molecules/TimedToast.tsx

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import React, { useEffect, useRef } from 'react'
+import { useLocation } from 'react-router-dom'
+import styled, { useTheme } from 'styled-components'
+
+import { faCheck, faTimes } from 'lib-icons'
+
+import { IconOnlyButton } from '../atoms/buttons/IconOnlyButton'
+import { tabletMin } from '../breakpoints'
+import { FixedSpaceRow } from '../layout/flex-helpers'
+import { modalZIndex } from '../layout/z-helpers'
+import { defaultMargins } from '../white-space'
+
+export interface Props {
+  onClick?: () => void
+  onClose: () => void
+  children?: React.ReactNode
+  'aria-label'?: string
+  'data-qa'?: string
+  closeLabel: string
+}
+
+const defaultDuration = 10000
+
+export default React.memo(function TimedToast({
+  onClick,
+  onClose,
+  children,
+  'aria-label': ariaLabel,
+  'data-qa': dataQa,
+  closeLabel
+}: Props) {
+  const { colors } = useTheme()
+  const timerRef = useRef<ReturnType<typeof setTimeout> | null>(null)
+  const location = useLocation()
+  const initialUrlPath = useRef(location.pathname)
+
+  const stopTimer = () => {
+    if (timerRef.current) {
+      clearTimeout(timerRef.current)
+    }
+  }
+
+  useEffect(() => {
+    timerRef.current = setTimeout(() => {
+      onClose()
+    }, defaultDuration)
+    return () => {
+      if (timerRef.current) {
+        clearTimeout(timerRef.current)
+      }
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Olisiko tässä parempi antaa tuo dependency, kuin disabloida linter rule?
```suggestion
  }, [onClose])
```